kit/media/src/UIVideo/hooks.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import type { Dimensions } from './types';
+
+export const useDimensionsByAspectRatio = (
+    width: number | undefined,
+    height: number | undefined,
+    aspectRatio: number | undefined,
+): Dimensions | null => {
+    return React.useMemo(() => {
+        if (!width && !height) {
+            return null;
+        }
+        if (!aspectRatio) {
+            return { width, height };
+        }
+        if (!width && height) {
+            return {
+                width: height * aspectRatio,
+                height,
+            };
+        }
+        if (width && !height) {
+            return {
+                width,
+                height: width / aspectRatio,
+            };
+        }
+        if (width && height) {
+            if (height > width / aspectRatio) {
+                return {
+                    width,
+                    height: width / aspectRatio,
+                };
+            }
```

Comment:
Very small optimization suggestion :)
```suggestion
            const aspectedHeight = width / aspectRatio;
            if (aspectedHeight < height) {
                return {
                    width,
                    height: aspectedHeight,
                };
            }
```

P.S. I was thought that `<` is always better than `>` in terms of Assembly optimization, just an interesting fact ;)